client/app/src/main/java/com/sinbaram/mapgo/SnsFeedActivity.kt

```diff
@@ -0,0 +1,234 @@
+                    recycler.layoutManager = LinearLayoutManager(this@SnsFeedActivity)
+                }
+            }
+
+            override fun onFailure(call: Call<List<Comment>>, t: Throwable) {
+                Log.d("respError", t.toString())
+            }
+        })
+    }
+
+    /** upload comment to post
+     * @param postId : id of post which want to upload comments to
+     * @param userId : id of user which will be author of comment
+     * @param comment : content of comment
+     * */
+    fun uploadPostComment(postId : Int, userId : Int, comment : String) {
+        val apiCall : Call<String> = serverAPI.PostComment(postId, userId, comment)
+        apiCall.enqueue(object : Callback<String> {
+            override fun onResponse(call: Call<String>, response: Response<String>) {
+                Log.d("test", "comment done")
+                getPostComment(postId)
+            }
+
+            override fun onFailure(call: Call<String>, t: Throwable) {
+                Log.d("respError", t.toString())
+            }
+        })
+    }
+
+    /** add like to post
+     * @param postId : id of post which added like to
+     * @param userId : id of user who liked to this post
+     */
+    fun setPostLike(postId : Int, userId : Int) {
+        val apiCall : Call<String> = serverAPI.AddLike(postId, userId)
+        apiCall.enqueue(object : Callback<String> {
+            override fun onResponse(call: Call<String>, response: Response<String>) {
+                Log.d("test", "add like done")
+                likerList+=userId
+            }
+
+            override fun onFailure(call: Call<String>, t: Throwable) {
+                Log.d("respError", t.toString())
+            }
+        })
+    }
+
+    /** remove like to post
+     * @param postId : id of post which removed like to
+     * @param userId : id of user who unliked to this post
+     */
+    fun removePostLike(postId : Int, userId : Int) {
+        val apiCall : Call<String> = serverAPI.DeleteLike(postId, userId)
+        apiCall.enqueue(object : Callback<String> {
+            override fun onResponse(call: Call<String>, response: Response<String>) {
+                Log.d("test", "remove like done")
+                likerList.remove(userId)
+            }
+
+            override fun onFailure(call: Call<String>, t: Throwable) {
+                Log.d("respError", t.toString())
+            }
+        })
+    }
+}
+
+/** class of informations of sns feed */
+class SnsFeedInformation(postInfo: PostFeedItem, baseurl : String) {
+    val postId: Int = postInfo.postID
+    val writerName: String = postInfo.writer.username
+    val writerImage: String = baseurl + postInfo.writer.picture
+    val postcontent: String = postInfo.contents
+    val postTime: String = postInfo.postTime.split(".")[0].replace("T", " ")
+    val postLat: Double = postInfo.location.lat
+    val postLong: Double = postInfo.location.lng
+    val liker = postInfo.like
+    var likerCount: Int = 0
+}
```

Comment:
```suggestion
/** class of informations of sns feed */
data class SnsFeedInformation(
    val postId: Int,
    val writerName: String,
    val writerImage: String,
    val postcontent: String,
    val postTime: String,
    val postLat: Double,
    val postLong: Double,
    val liker: List<Like>,
    var likerCount: Int,
    ) {
    constructor(postInfo: PostFeedItem, baseurl : String) : this(
        postInfo.postID,
        postInfo.writer.username,
        baseurl + postInfo.writer.picture,
        postInfo.contents,
        postInfo.postTime.split(".")[0].replace("T", " "),
        postInfo.location.lat,
        postInfo.location.lng,
        postInfo.like,
        0,
    )
}
```
This kind of class have an advantage when declare it as a data class, as it kvm automatically add equals, 
hashCode, etc.. Although it may seems little bit more verbose, I think this is a more kotlinic way.

Reply:
Thanks for your suggestion, I'll commit this right away!